packages/indexer-common/src/indexer-management/actions.ts

```diff
@@ -12,13 +15,61 @@ import { Eventual, join, Logger, timer } from '@graphprotocol/common-ts'
 export class ActionManager {
   constructor(
     public allocationManager: AllocationManager,
+    public networkMonitor: NetworkMonitor,
+    private currentEpoch: number,
+    private maxAllocationEpoch: number,
     private logger: Logger,
     private models: IndexerManagementModels,
+    private allocationManagementMode?: AllocationManagementMode,
+    private autoMinBatchSize?: number,
+    private autoMaxWaitTime?: number,
   ) {}
 
+  private async batchReady(
+    approvedActions: Action[],
+    pollingFrequency: number,
+  ): Promise<boolean> {
+    if (approvedActions.length < 1) {
+      return false
+    }
+
+    // In AUTO mode, worker waits to execute when 1) the batch is not filled upto minimum size
+    // and 2) oldest allocatation is not expiring after the current epoch
+    // and 3) oldest action is not stale (10 times the user provided threshold of polling frequency)
+    if (this.allocationManagementMode === AllocationManagementMode.AUTO) {
+      const oldestCreatedAtEpoch = (
+        await this.networkMonitor.allocations(AllocationStatus.ACTIVE)
+      )[0].createdAtEpoch
+
+      const updatedDeadline = new Date(new Date().valueOf() - 10 * pollingFrequency)
```

Comment:
Not sure if this relationship makes sense with L66 and 68. I didn't want to add another start-up parameter, but I can see how the 2 time thresholds shouldn't be related